core/ts/MaciState.ts

```diff
@@ -1361,11 +1360,15 @@ class Poll {
      * @param maciState the reference to the MaciState Class
      * @returns a new Poll instance
      */
-    static fromJSON(json: any, maciState: MaciState): Poll {
+    static fromJSON(
+        json: any, 
+        maciState: MaciState, 
+        coordinatorPrivateKey: string
+    ): Poll {
```

Comment:
@baumstern  are you suggesting we reconstruct the object without a key (by passing a random one) and then set it after `fromJSON` finishes, and before doing any processing? this could work

Reply:
@baumstern reflected in the latest commits

Reply:
> @baumstern are you suggesting we reconstruct the object without the actual valid key (by passing a random one) and then set it after `fromJSON` finishes, and before doing any processing? this could work

Thank you for incorporating the feedback. While using `new Keypair()` in fromJSON and setting the `coordinatorKeypair` later does technically work, it introduces a requirement for users to be aware of an additional step (`setCoordinatorKeypair()`) to fully initialize the `Poll` object. This could potentially lead to confusion or errors if the user is not aware that the `Poll` instance is incomplete after `fromJSON`.

Reply:
I think fully addressing this issue is pretty tricky and I want to avoid prolonging this PR too much, as it is not main focus of this PR. We could handle this matter separately, possibly in a dedicated issue or follow-up PR. How would you like to proceed?

Reply:
Also..should we merge this PR first or wait until after merging the cli refactoring PR?

Reply:
@baumstern the only other solution I can think of, which could be much easier and clear, is to actually save the coordinator key in the serialized object - `toJSON` only converts to JSON, it does not save to disk. However, when used in the cli, before saving to disk, we remove the privateKey entry from the object, save it without, and while deserializing, we amend the json object created extracted from the local file with the actual key (which would be an input anyways to `genProofs`)

Reply:
> Also..should we merge this PR first or wait until after merging the cli refactoring PR?

I think we could merge this first while I work on the feedback of the CLI one, then incorporate this one command into the new cli. lmk what you think

Reply:
> @baumstern the only other solution I can think of, which could be much easier and clear, is to actually save the coordinator key in the serialized object - technically speaking `toJSON` only converts to JSON, it does not save to disk (it is up to the user to actually store it on disk later if they wish so). However, when used in the cli, before saving to disk, we remove the privateKey entry from the object, save it without, and while deserializing, we amend the json object created extracted from the local file with the actual key (which would be an input anyways to `genProofs`)

Sounds like a hacky solution haha. It's also technically make sense...but..since `toJSON` is kinda universal method so actually we can't prevent user to export their coordinator key (maybe accidently). We can remove coordinator key before write JSON into disk in our cli implementation but it is possible that third-party application that importing our package and exporting actual coordinator key into their disk. So I think it could be seen as a making attack space(maybe precisely making space that user could make mistake). Since coordinator key is definetely sensitive information, we should be careful and minimize the security risk.

Reply:
> > Also..should we merge this PR first or wait until after merging the cli refactoring PR?
> 
> I think we could merge this first while I work on the feedback of the CLI one, then incorporate this one command into the new cli. lmk what you think

Sounds good!